### ISSUE TITLE
Travis CI needs to be turned on

### ISSUE THREAD

**Issue by cclauss** [sherlock-project/sherlock#235]
The administrator of this repo would need to log into https://travis-ci.com/sherlock-project with their GitHub credentials and turn this free automated testing service back on like it used to work at https://travis-ci.com/TheYahya/sherlock/pull_requests

**Comment by hoadlck** [comment]
There are so many sites that are unreliable, it is pretty uncommon to be able to run the tests without one or the other failing to respond and causing the tests to fail.

Unfortunately, I do not have the bandwidth to think thru how to implement how I think the tests should work.  But, it seems that there are two broad areas:
1. Detection Algorithms
Ensure that the nuts and bolts of the detection logic actually works.
1. Site Integration
Detect if any site is not giving expected results with Sherlock.  This happens when a site changes its behavior, and Sherlock give a false positive.

Does anyone have ideas on how point 2 would work?  I am thinking that it would be nice to have a list of all of the sites, and a status next to each: Passed, Failed, Error Connecting.  The test that would be meaningful would be more like a dashboard.  While it is straightforward enough to get these test results when running locally, is there a way to display more subtle test results using Travis?  Does anyone have any examples or experience in doing things like this?

**Comment by cclauss** [comment]
Thanks for merging #234 because I believe that now unittest provides the Passed, Failed, Error Connecting report that you request above.  However, now it runs __unittest || true__ which means that __Travis remains green__ even if one or more unittests fail.  You have your report but the unittest command never fails the Travis job.  Best of Both Worlds (BoBW).

**Comment by cclauss** [comment]
If we can figure out different tests for 1. and 2. above then we could create two parallel Travis CI runs.  One that tests sites for 1. but runs in __allow_failures__ mode and the second that tests sites for 2. that will fail the Travis job if any of its unittests fails.  I will give the test some thought...

**Comment by hoadlck** [comment]
Well, I actually do have the tests split out so that there are tests for the two different situations.

I made PR #248 which does Part 1.  I changed the CI to only do
```
python -m unittest tests.all.SherlockDetectTests --buffer --verbose
```
If you wanted to do all of the coverage tests, then the command would be
```
python -m unittest tests.all.SherlockSiteCoverageTests --buffer --verbose
```
I am not sure how you would run the tests in parallel.

What would really be ideal would be to have the last tested status for each site in the [sites.md](https://github.com/sherlock-project/sherlock/blob/master/sites.md).  This would be a natural dashboard.

**Comment by hoadlck** [comment]
@TheYahya Are you able to turn the CI back on?  It was working at one time.  The comment at the beginning of this issue has the link to log into.

**Comment by hoadlck** [comment]
The CI does seem to be turned on. I am not seeing e-mails from the process, but I do see the status updating.

**Comment by cclauss** [comment]
Not turned on at https://travis-ci.com/sherlock-project or https://travis-ci.org/sherlock-project

**Comment by hoadlck** [comment]
The status in the readme.md is pointing at https://travis-ci.com/TheYahya/sherlock/.  So, it looks like @TheYahya has this pointed to the earlier place.  @TheYahya can you turn it on and update the readme to point to the new location?

**Comment by paulodiovani** [comment]
Please, turn travis-ci on for https://travis-ci.com/sherlock-project/sherlock/

I can update the readme after that.

**Comment by hoadlck** [comment]
We have converted over to GitHub actions.  We now have a unique test for Pull Requests and for updates to master.  I just added a nightly test for all of the sites as well.

I think it would be good to have the nightly build automatically add an issue to fix the sites that fail.  As it is, we get a scattering of notifications from users when they are focused on a certain site.

But, this issue has been handled.
